auth0/src/main/java/com/auth0/android/authentication/storage/SecureCredentialsManager.kt

```diff
@@ -303,9 +302,21 @@ public class SecureCredentialsManager @VisibleForTesting(otherwise = VisibleForT
             decryptCallback = null
             return
         }
-        val credentials = gson.fromJson(json, Credentials::class.java)
+        val bridgeCredentials = gson.fromJson(json, OptionalCredentials::class.java)
+        /* OPTIONAL CREDENTIALS
+         * This bridge is required to prevent users from being logged out when
+         * migrating from Credentials with optional Access Token and ID token
+         */
+        val credentials = Credentials(
+            bridgeCredentials.idToken.orEmpty(),
```

Comment:
handles migration from "legacy credentials" passing empty strings when values were not previously saved